app/scripts/migrations/097.test.ts

```diff
@@ -0,0 +1,103 @@
+import { migrate, version } from './097';
+
+const oldVersion = 96;
+describe('migration #97', () => {
+  it('updates the version metadata', async () => {
+    const oldStorage = {
+      meta: { version: oldVersion },
+      data: {},
+    };
+
+    const newStorage = await migrate(oldStorage);
+
+    expect(newStorage.meta).toStrictEqual({ version });
+  });
+
+  it('handles missing TransactionController', async () => {
+    const oldState = {
+      OtherController: {},
+    };
+
+    const transformedState = await migrate({
+      meta: { version: oldVersion },
+      data: oldState,
+    });
+
+    expect(transformedState.data).toEqual(oldState);
+  });
+
+  it('handles empty transactions', async () => {
+    const oldState = {
+      TransactionController: {
+        transactions: {},
+      },
+    };
+
+    const transformedState = await migrate({
+      meta: { version: oldVersion },
+      data: oldState,
+    });
+
+    expect(transformedState.data).toEqual(oldState);
+  });
+
+  it('handles missing state', async () => {
+    const transformedState = await migrate({
+      meta: { version: oldVersion },
+      data: {},
+    });
+
+    expect(transformedState.data).toEqual({});
+  });
+
+  it('removes nonceDetail from transactions', async () => {
+    const oldState = {
+      TransactionController: {
+        transactions: {
+          tx1: {
+            nonceDetails: {
```

Comment:
Should we add a second transaction with `nonceDetails` just to ensure we're iterating correctly and supporting multiple?

Reply:
Done

Reply:
Done